### PR TITLE
feat(kuma-cp) dataplane token boundary improvements

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
This PR improves Dataplane Token so instead of generating it for mandatory mesh+name you can generate it by
1) nothing, so you can use it across all DPs and Meshes
2) Mesh, so you can use it across all DPs and given mesh
3) Mesh+Name - current behavior
4) Mesh+Tag, so you can use it across all instances of one service

### Documentation

- [ ] TODO

### CODE REVIEW

File: pkg/sds/auth/universal/auth_test.go [hunk 1]
```diff
@@ -49,90 +42,153 @@ var _ = Describe("Authentication flow", func() {
...
-		// then
-		Expect(err).ToNot(HaveOccurred())
-		Expect(authIdentity.Services[0]).To(Equal("web"))
-		Expect(authIdentity.Mesh).To(Equal(id.Mesh))
 	})
 
-	It("should throw an error on invalid token", func() {
-		// when
-		id := xds.ProxyId{
+	DescribeTable("should correctly authenticate dataplane",
```
Comment (nickolaev) [inline pkg/sds/auth/universal/auth_test.go:65]: Awesome!

File: pkg/sds/auth/universal/auth_test.go [hunk 2]
```diff
@@ -49,90 +42,153 @@ var _ = Describe("Authentication flow", func() {
...
-			Name: "dp1",
-		}
-		token, err := issuer.Generate(generateId)
-
-		// then
-		Expect(err).ToNot(HaveOccurred())
+			Tags: map[string]map[string]bool{
+				"kuma.io/service": {
+					"web":     true,
+					"web-api": true,
```
Comment (nickolaev) [inline pkg/sds/auth/universal/auth_test.go:101]: Interesting, can we do these? How to I express this in curl?

File: pkg/sds/auth/universal/authenticator.go [hunk 3]
```diff
@@ -18,34 +20,66 @@ func NewAuthenticator(issuer builtin_issuer.DataplaneTokenIssuer, dataplaneResol
...
 }
 
+// universalAuthenticator defines authentication for Dataplane Tokens
+// All fields in token are optional, so we only validate data that is available in token. This way you can pick your level of security.
+// Generate token for mesh+name for maximum security.
+// Generate token for mesh+tags(ex. kuma.io/service) so you can reuse the token for many instances of the same service.
+// Generate token for mesh if you trust the scope of the mesh.
+//
+// If you generate token bound to tags all tags values have to match the dataplane, so for example if you have a Dataplane
+// with inbounds: 1) kuma.io/service:web 2) kuma.io/service:web-api, you need token for both values kuma.io/service=web,web-api
```
Comment (nickolaev) [inline pkg/sds/auth/universal/authenticator.go:30]: Interesting, can I do "wildcards" here?

**Review by nickolaev — APPROVED** [review]
Impressive! Looks very elegant. We definitely need docs for this though!

File: pkg/sds/auth/universal/auth_test.go (hunk 2 above)
Comment (jakubdyszkiewicz) [reply pkg/sds/auth/universal/auth_test.go:101]:
```
{
  "tags": {
    "kuma.io/service": ["web", "web-api"]
  }
}
```

File: pkg/sds/auth/universal/authenticator.go (hunk 3 above)
Comment (jakubdyszkiewicz) [reply pkg/sds/auth/universal/authenticator.go:30]: Wildcard in the form of `kuma.io/service: *` does not make sense. You can just omit the tag.
Wildcard in the form of `kuma.io/service: web*` makes sense, but I'm hesitant to introduce it now. I'd introduce it if there is a real need for it.

**Pull request merged**